bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBestpriceHandler.java

```diff
@@ -0,0 +1,267 @@
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.awattar.internal.AwattarBestPriceResult;
+import org.openhab.binding.awattar.internal.AwattarBestpriceConfiguration;
+import org.openhab.binding.awattar.internal.AwattarConsecutiveBestPriceResult;
+import org.openhab.binding.awattar.internal.AwattarNonConsecutiveBestPriceResult;
+import org.openhab.binding.awattar.internal.AwattarPrice;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AwattarBestpriceHandler} is responsible for computing the best prices for a given configuration.
+ *
+ * @author Wolfgang Klimt - Initial contribution
+ */
+@NonNullByDefault
+public class AwattarBestpriceHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AwattarBestpriceHandler.class);
+
+    private final int thingRefreshInterval = 60;
+    @Nullable
+    private ScheduledFuture<?> thingRefresher;
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    public AwattarBestpriceHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        AwattarBestpriceConfiguration config = getConfigAs(AwattarBestpriceConfiguration.class);
+
+        boolean configValid = true;
+
+        if (config.length >= config.rangeDuration) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/error.length.value");
+            configValid = false;
+        }
+
+        if (!configValid) {
+            return;
+        }
+
+        synchronized (this) {
+            ScheduledFuture<?> localRefresher = thingRefresher;
+            if (localRefresher == null || localRefresher.isCancelled()) {
+                /*
+                 * The scheduler is required to run exactly at minute borders, hence we can't use scheduleWithFixedDelay
+                 * here
+                 */
+                thingRefresher = scheduler.scheduleAtFixedRate(this::refreshChannels,
+                        getMillisToNextMinute(1, timeZoneProvider), thingRefreshInterval * 1000, TimeUnit.MILLISECONDS);
+            }
+
```

Comment:
This empty line needs to be removed to satisfy checkstyle.

Reply:
Removed it, however, There was no warning when I built the binding locally. I will double check the build result after push.